src/jobs/jobs.controller.ts

```diff
@@ -475,18 +476,98 @@ export class JobsController {
     @Req() request: Request,
     @Body() createJobDto: CreateJobDto,
   ): Promise<JobClass> {
-    Logger.log("creating job!")
+    Logger.log("Creating job!");
     // Validate that request matches the current configuration
-    await this.validateJob(createJobDto, request);
+    await this.validateJob(createJobDto);
     // Check job authorization
-    await this.instanceAuthorization(createJobDto, request.user as JWTUser);
+    await this.instanceAuthorization(createJobDto.jobParams, request.user as JWTUser);
     // Create actual job in database
     const createdJobInstance = await this.jobsService.create(createJobDto);
     // Perform the action that is specified in the create portion of the job configuration
-    const jobServiceResponse = await this.performJobCreateAction(createdJobInstance);
+    await this.performJobCreateAction(createdJobInstance);
+    // TBD
     // Update job instance with results of job create action
-    // return await this.jobsService.update(jobServiceResponse);
-    return jobServiceResponse; // or createdJobInstance ?
+    // return await this.jobsService.statusUpdate(createdJobInstance);
+    return createdJobInstance;
+  }
+
+
+  /**
+   * Update job
+   */
+  @UseGuards(PoliciesGuard)
+  @CheckPolicies((ability: AppAbility) =>
+    ability.can(AuthOp.JobsUpdate, JobClass),
+  )
+  @Patch(":id")
+  @ApiOperation({
+    summary: "It updates an existing job.",
+    description: "It updates an existing job.",
+  })
+  @ApiBody({
+    description: "Input fields for the job to be updated",
+    required: true,
+    // schema: new UpdateJobStatusDto, // TBD
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: JobClass,
+    description: "Updated job",
+  })
+  async update(
+    @Req() request: Request,
+    @Param("id") id: string,
+    @Body() updateJobDto: UpdateJobStatusDto,
+  ): Promise<JobClass | null> {
+    Logger.log("updating job!");
+    // Find existing job
+    const currentJob = await this.jobsService.findOne({ id: id });
+    if (currentJob === null) {
+      throw new HttpException(
+        {
+          status: HttpStatus.BAD_REQUEST,
+          message: "Invalid job id.",
+        },
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    // Check job authorization
+    await this.instanceAuthorization(currentJob.jobParams, request.user as JWTUser);
+    // Update job in database
+    const updatedJob = await this.jobsService.statusUpdate(id, updateJobDto);
```

Comment:
Currently `update()` only performs a **status update** (only `statusMessage`, not `statusCode`). 
1. Is there a need to update more fields? If yes, which ones, why and when? 
2. Do we need to perform an update after creation is complete? And is it just a status update? If yes, what would the new job status be? If no, which other fields do we also need to update?
3. Which are the acceptable values for `statusMessage`? What are their corresponding `statusCode` values?

Reply:
Status update should be able to update the status message and status code.

Every time the system receives an update request, it should update the statusMessage and statusCode fields, together with append them to the update history.

Job updates will be performed until the job is completed successfully or not. 
The only thing that we can do is status update, we are not allowed to change anything else.

statusMessages are just arbitrary strings. statusCode should be defined in the job definition... at least that was my idea.

Reply:
- `update()` should perform a status update (`statusMessage` and `statusCode`)
- We will not impose any type-checking.
- After a successful creation, a job get updated with these values: `{ "statusMessage": "Job has been created", "statusCode": "jobCreated" }`

Reply:
Perfect!!!